imagepipeline/src/main/java/com/facebook/imagepipeline/producers/ResizeAndRotateProducer.java

```diff
@@ -237,19 +256,21 @@ private static TriState shouldTransform(
     final float heightRatio = ((float) resizeOptions.height) / height;
     float ratio = Math.max(widthRatio, heightRatio);
 
-    // TODO: The limit is larger than this on newer devices. The problem is to get the real limit,
-    // you have to call Canvas.getMaximumBitmapWidth/Height on a real HW-accelerated Canvas.
-    if (width * ratio > BitmapUtil.MAX_BITMAP_SIZE) {
-      ratio = BitmapUtil.MAX_BITMAP_SIZE / width;
-    }
-    if (height * ratio > BitmapUtil.MAX_BITMAP_SIZE) {
-      ratio = BitmapUtil.MAX_BITMAP_SIZE / height;
-    }
+    // TODO: Both width and height should be equal or smaller than device's max bitmap size
+    float maxBitmapSize = Math.max(resizeOptions.width, resizeOptions.height);
+
+    if (width * ratio > maxBitmapSize) {
+	      ratio = maxBitmapSize / width;
+	    }
+	    if (height * ratio > maxBitmapSize) {
+	      ratio = maxBitmapSize / height;
+	    }
+    FLog.v(TAG, "Resize ratio: %f", ratio);
     return ratio;
   }
 
   @VisibleForTesting static int roundNumerator(float maxRatio) {
-    return (int) (ROUNDUP_FRACTION + maxRatio * JpegTranscoder.SCALE_DENOMINATOR);
```

Comment:
We should keep this, but possibly make it configurable. There is a reason why we added this.
Without it, we would always round down which is not desireable. We intentionally added this offset, so that the x in range [4.333, 5.333] gets round to 5 and so on.